src/KUSignalLib/DSP/PLL.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+class PLL():
+    LFK2prev = 0
+    phase = 0
+
+    def __init__(self, kp=1, k1=0, k2=0, wstart = 1, fs = 1):
+        self.Kp = kp
+        self.K1 = k1
+        self.K2 = k2
+        self.w0 = wstart
+        self.fs = fs
+
+    def insertNewSample(value):
+        """
+        insert a new sample of the received signal we are trying to lock to.
+        """
+        pass
+
+    def phaseDetector(self, sample1, sample2, Kp = None):
+        """
+        Phase detector implementation.
+
+        :param sample1: Complex number. First point.
+        :param sample2: Complex number. Second point.
+        :param Kp: Float type. Proportional gain, should be less than one.
+        :return: Float type. Phase difference between the two points.
+        """
+        if Kp is None:
```

Comment:
maybe don't want to check this everytime and just assume inheritance.... i dont think we would ever change it "mid-PLL"